torch/distributed/_state_dict_utils.py

```diff
@@ -585,6 +596,34 @@ def _broadcast_state_dict(
         _broadcast_tensors(ret, local_state_dict, keys, device, pg)
 
 
+def _distribute_state_dict(
+    full_state_dict: Dict[str, Any],
+    local_state_dict: Dict[str, Any],
+    device: torch.device,
+    pg: Optional[dist.ProcessGroup] = None,
+) -> None:
+    # Distribute tensors in each rank
+    ret = {}
+    for key, value in full_state_dict.items():
+        if not torch.is_tensor(value) and key in local_state_dict:
+            local_state_dict[key] = value
+        elif value.dim() == 0 and key in local_state_dict:
+            local_state_dict[key] = value.cpu()
+        else:
+            ret[key] = _TensorInfo(value.size(), value.dtype)
```

Comment:
Why do you need this? It is not used in any other place.

Reply:
Thank you for the comment, I will remove ret in the next version